plugins/output/flight/flight.go

```diff
@@ -47,27 +48,86 @@ func (f *Flight) Write(metrics []telegraf.Metric) error {
 	builder := array.NewRecordBuilder(memory.DefaultAllocator, &f.schema)
 	defer builder.Release()
 
-	// Currently, the plugin only implements support for the simplest schema
-	// supported by legacy JVM and current Rust versions of [ModelarDB](https://github.com/ModelarData/ModelarDB-RS),
-	// as listed below. Support for an arbitrary schema is planned.
-
-	tidBuilder := builder.Field(0).(*array.Int32Builder) // Unused and deprecated time series identifier.
-	timeBuilder := builder.Field(1).(*array.TimestampBuilder)
-	valueBuilder := builder.Field(2).(*array.Float32Builder)
+	schemaFields := f.schema.Fields()
 
 	builder.Reserve(len(metrics))
 
 	for _, m := range metrics {
 
-		tidBuilder.AppendValues([]int32{1}, nil)
-
 		timeInt := m.Time().UnixMilli()
 
-		timeBuilder.AppendValues([]arrow.Timestamp{arrow.Timestamp(timeInt)}, nil)
-
-		fieldlst := m.FieldList()
-
-		valueBuilder.AppendValues([]float32{float32(fieldlst[0].Value.(float64))}, nil)
+		for i, f := range schemaFields {
+			switch f.Type.ID() {
+			case arrow.TIMESTAMP:
+				builder.Field(i).(*array.TimestampBuilder).AppendValues([]arrow.Timestamp{arrow.Timestamp(timeInt)}, nil)
+			case arrow.STRING:
+				tag, wasSet := m.GetTag(f.Name)
+				if !wasSet {
+					fmt.Printf("tag %d : %s not set", i, f.Name)
+				}
+				builder.Field(i).(*array.StringBuilder).AppendValues([]string{tag}, nil)
+			case arrow.INT32:
+				field, wasSet := m.GetField(f.Name)
+				if !wasSet {
+					fmt.Printf("field %d : %s not set", i, f.Name)
+				}
+				switch v := field.(type) {
```

Comment:
You are right, Go supports multiple case values. But the switch here is a type switch, that in some ways does not support multiple case values. When a type assertion is made, it is only used on the first case in the multiple cases entered, but the rest of the cases are interfaces. As I cannot insert an interface into the builder, I don't see a workaround for this. For more information, see [this](https://stackoverflow.com/questions/40575033/golang-multiple-case-in-type-switch).

![image](https://user-images.githubusercontent.com/43062786/187420724-9dd7cd28-a95b-46ba-a57f-e9dc0d2c1b9e.png)

Reply:
Could you use an if statement instead? This would make it possible to have multiple boolean expressions in a single check to group the duplicate case handlers.

Reply:
As mentioned in this [comment](https://github.com/ModelarData/Telegraf-Output-Apache-Arrow-Flight/pull/2#discussion_r958565365) maybe `go generate` could be used to reduce the amount of duplicate code.

Reply:
That is true, I would be able to use an if statement here. The problem is that I have to do a type assertion to extract the value from the interface, and I can't do three separate type assertions within one if else statement. This is exactly why type switches were [made](https://stackoverflow.com/questions/6372474/how-to-determine-an-interface-values-real-type/6373426#6373426).

![image](https://user-images.githubusercontent.com/43062786/187499773-bfa3072f-7cc5-4a8b-891a-d36bb34f8017.png)

As you can see in this image, I have no idea how I would approach the type in the question marks.